src/components/SpaceCreator.tsx

```diff
@@ -52,7 +55,21 @@ export function SpaceCreatorForm({
 
     setSubmitted(true)
     try {
-      const space = await client.createSpace(name)
+
+      const gatewayId = (process.env.NEXT_PUBLIC_W3UP_GATEWAY_ID || 'did:web:w3s.link') as DID<'web'>
+      const gatewayUrl = process.env.NEXT_PUBLIC_W3UP_GATEWAY_HOST || 'https://freeway.dag.haus' as string
+
+      const storachaGateway = UcantoClient.connect({
+        id: {
+          did: () => gatewayId
+        },
+        codec: CAR.outbound,
+        channel: HTTP.open({ url: new URL(gatewayUrl) }),
+      }) as ConnectionView<ContentServeService>
+
+      const space = await client.createSpace(name, {
+        authorizeGatewayServices: [storachaGateway]
```

Comment:
**issue:** TypeScript also says this isn't a property here. Is that just the need for an upgrade, too?